macosx/PrefsController.m

```diff
@@ -845,7 +845,8 @@ - (void) folderSheetShow: (id) sender
             [fDefaults setBool: YES forKey: @"DownloadLocationConstant"];
             [self updateShowAddMagnetWindowField];
             
-            tr_sessionSetDownloadDir(fHandle, [folder UTF8String]);
+            assert(folder.length > 0);
```

Comment:
I guess the question is, are you checking post-condition for panel or pre-condition for libtransmission. For the former (since you say assertion is unlikely to fail specifically in regards to the panel), I'd be okay with leaving the assertion here but moving it closer to `folder` initialization (or otherwise farther from call into libtransmission). For the latter, I think moving the assertion into the libtransmission would be more appropriate: this avoids possible further duplication in other places, plus makes it easier to see the restrictions if we decide to introduce real argument checks later on. This applies to other two calls below as well.